### PR TITLE
Correct parameter documentation

### PR DESCRIPTION
**Pull request by johnnonweiler**
Someone previously ran a script making changes to 1,207 files, including changing old function documentation into a doxygen format.  In java_local_variable_table.cpp,  the script failed and messed up the comments.  This commit fixes the faulty automatic reformatting, and also adds documentation for missing parameters.

(This fixes a few doxygen warnings, as well as many cases where the doxygen documentation was wrong but doxygen wasn't generating any warnings.)

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] My contribution is formatted in line with CODING_STANDARD.md.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: jbmc/src/java_bytecode/java_local_variable_table.cpp [hunk 1]
```diff
@@ -635,8 +639,8 @@ void java_bytecode_convert_methodt::find_initializers_for_slot(
 
 /// Walk a vector, a contiguous block of entries with equal slot index at a
 /// time.
-/// \par parameters: `it1` and `it2`, which are iterators into the same vector,
-/// of which `itend` is the end() iterator.
+/// `it1` and `it2` are iterators into the same vector, of which `itend` is the
+///   end() iterator.
 /// \return Moves `it1` and `it2` to delimit a sequence of variable table
```
Comment (smowton) [inline jbmc/src/java_bytecode/java_local_variable_table.cpp]: Shouldn't use `\return` for a void function

File: jbmc/src/java_bytecode/java_local_variable_table.cpp [hunk 2]
```diff
@@ -683,7 +687,7 @@ void java_bytecode_convert_methodt::find_initializers(
 }
 
 /// See above
-/// \par parameters: `vars_with_holes`: variable table
+/// \param vars_with_holes: variable table
 /// \return Removes zero-size entries from `vars_with_holes`
```
Comment (smowton) [inline jbmc/src/java_bytecode/java_local_variable_table.cpp]: Again `\return` for void

File: jbmc/src/java_bytecode/java_local_variable_table.cpp [hunk 3]
```diff
@@ -708,8 +712,8 @@ static void cleanup_var_table(
 }
 
 /// See `find_initializers_for_slot` above for more detail.
-/// \par parameters: `m`: Java method
-/// `amap`: Map from bytecode indices to instructions in `m`
+/// \param m: Java method
+/// \param amap: Map from bytecode indices to instructions in `m`
 /// \return Populates `this->vars_with_holes` equal to
```
Comment (smowton) [inline jbmc/src/java_bytecode/java_local_variable_table.cpp]: \return for void

**Review by smowton — APPROVED** [review]
Looks good except I think for void-typed functions with side-effects we shouldn't be using the \return` tag

**Comment by johnnonweiler** [comment]
This is now updated so the \return tag is not used for any void functions.

File: jbmc/src/java_bytecode/java_local_variable_table.cpp [hunk 4]
```diff
@@ -277,6 +277,8 @@ static void populate_variable_address_map(
 ///   method
 /// \param [out] predecessor_map:
 ///   the output of the function, populated as described above
+/// \param msg_handler:
+///   for reporting warnings
```
Comment (owen-mc-diffblue) [inline jbmc/src/java_bytecode/java_local_variable_table.cpp:281]: No need for a line break after the colon (and the param above as well)

**Review by owen-mc-diffblue — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 140128e1e5f43202436420ad0811a7c343c189b4).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87660250
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 637610f71b3881f9aa399a35b1cdbae132eea1b8).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87661816

**Pull request merged**